samples/estimation/df-chemistry/chemistry.py

```diff
@@ -0,0 +1,482 @@
+                # so accumulate the whole header first and then parse.
+                header += line + ' '
+                if line.endswith("&END"):
+                    # Strip "&FCI" and "&END"
+                    rest_header = header[4:-4]
+
+                    # Find each key-value pair based on the "=" sign
+                    while (ind := rest_header.find('=')) > -1:
+                        key = rest_header[0:ind].strip()
+                        rest_header = rest_header[ind + 1:]
+                        # Figure out which part of the rest is value: until
+                        # either whitespace or (if key is not ORBSYM) comma
+                        ind = rest_header.find(',' if key != "ORBSYM" else ' ')
+                        value = rest_header[0:ind].strip()
+                        rest_header = rest_header[ind + 1:]
+                        if key == "ORBSYM":
+                            value = value[:-1]
+                        header_values[key] = value
+                    parse_header = False
+            else:
+                tokens = line.split()
+                coefficient = float(tokens[0])
+                indices = [int(str) - 1 for str in tokens[1:] if str != '0']
+
+                if len(indices) == 2:
+                    indices.sort()
+                    self.one_body_terms.append((indices, coefficient))
+                elif len(indices) == 4:
+                    # conversion from Mulliken to Dirac
+                    i = indices[0]
+                    j = indices[2]
+                    k = indices[3]
+                    l = indices[1]
+
+                    symmetries = [
+                        [i, j, k, l],
+                        [j, i, l, k],
+                        [k, l, i, j],
+                        [l, k, j, i],
+                        [i, k, j, l],
+                        [k, i, l, j],
+                        [j, l, i, k],
+                        [l, j, k, i],
+                    ]
+                    symmetries.sort()
+                    self.two_body_terms.append((symmetries[0], coefficient))
+
+        self.num_orbitals = int(header_values['NORB'])
+
+        return self
+
+
+@dataclass
+class TwoBodyResult:
+    eigenvalues: list[npt.NDArray[np.float64]]
+    eigenvectors: list[npt.NDArray[np.float64]]
+    one_norms: list[float]
+    two_norms: list[float]
+
+
+@dataclass
+class DoubleFactorization:
+    num_orbitals: int
+    rank: int
+    one_body_norm: float
+    two_body_norm: float
+    one_body_eigenvalues: list[float]
+    one_body_eigenvectors: npt.NDArray[npt.NDArray[np.float64]]
+    two_body_eigenvalues: npt.NDArray[npt.NDArray[np.float64]]
+    two_body_eigenvectors: list[npt.NDArray[npt.NDArray[np.float64]]]
+
+    @classmethod
+    def process_fcidump(self, structure: FCIDumpFileContent, error: float):
+
+        # The `structure` provides us with one- and two electron integrals
```

Comment:
Maybe we can mention the reference we are referring to in the beginning of the file or the class.

Reply:
Is there an official reference for the structure of the FCIDUMP file? I could not find one, only examples of different libraries reading it. The closest thing I could find was https://theochem.github.io/horton/2.0.2/user_hamiltonian_io.html#fcidump-format but it doesn't look like a solid reference